src/js/panels.js

```diff
@@ -16,20 +15,110 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+            // to this model property. The component is configured
+            // with the expectation that this is the salient model value.
+            value: null
+        },
+        range: {
+            min: 1,
+            max: 2
+        },
+        step: 0.1,
+        selectors: {
+            instructions: ".gpiic-fd-range-instructions",
+            meter: ".gpiic-fd-range-indicator",
+            increase: ".gpiic-fd-range-increase",
+            increaseLabel: ".gpiic-fd-range-increaseLabel",
+            decrease: ".gpiic-fd-range-decrease",
+            decreaseLabel: ".gpiic-fd-range-decreaseLabel"
+        },
+        selectorsToIgnore: ["meter", "increase", "decrease"],
+        protoTree: {
+            instructions: {messagekey: "instructions"},
+            increaseLabel: {messagekey: "increaseLabel"},
+            decreaseLabel: {messagekey: "decreaseLabel"}
+        },
+        invokers: {
+            stepUp: {
+                funcName: "gpii.firstDiscovery.panel.ranged.step",
+                args: ["{that}"]
+            },
+            stepDown: {
+                funcName: "gpii.firstDiscovery.panel.ranged.step",
+                args: ["{that}", true]
+            },
+            updateMeter: {
+                funcName: "gpii.firstDiscovery.panel.ranged.updateMeter",
+                args: ["{that}", "{that}.model.value"]
+            }
+        },
+        listeners: {
+            "afterRender.bindIncrease": {
+                "this": "{that}.dom.increase",
+                "method": "click",
+                "args": ["{that}.stepUp"]
+            },
+            "afterRender.bindDecrease": {
+                "this": "{that}.dom.decrease",
+                "method": "click",
+                "args": ["{that}.stepDown"]
+            },
+            "afterRender.updateMeter": "{that}.updateMeter"
+        },
+        modelListeners: {
+            "value": {
+                listener: "{that}.updateMeter",
+                excludeSource: ["init"]
             }
         }
     });
 
+    gpii.firstDiscovery.panel.ranged.clip = function (value, min, max) {
+        if (max > min) {
+            return Math.min(max, Math.max(min, value));
+        }
+    };
+
+    gpii.firstDiscovery.panel.ranged.step = function (that, reverse) {
+        var step = reverse ? (that.options.step * -1) : that.options.step;
+        var newValue = that.model.value + step;
+        newValue = gpii.firstDiscovery.panel.ranged.clip(newValue, that.options.range.min, that.options.range.max);
+        that.applier.change("value", newValue);
+    };
+
+    gpii.firstDiscovery.panel.ranged.calculatePercentage = function (value, min, max) {
+        if (max > min) {
+            var clipped = gpii.firstDiscovery.panel.ranged.clip(value, min, max);
+            return ((clipped - min) / (max - min)) * 100;
+        }
+    };
+
+    gpii.firstDiscovery.panel.ranged.updateMeter = function (that, value) {
```

Comment:
We need to make a call to `gpii.firstDiscovery.panel.ranged.calculatePercentage` in this function to determine what the new percentage should be. If we make this declarative we'll need to add this as an expander, but it's something that would have to be evaluated each time. I think it may be better not to use an expander for this particular case, so it may make more sense to leave it as a separate function for now.